docs/dependency_tracking.rst

```diff
@@ -0,0 +1,99 @@
+
+``rapids-cmake`` is designed to remove this dual source of truth by expanding the concept of Modern CMake `Usage Requirements <https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#build-specification-and-usage-requirements>`_  to include external packages.
+This is done via the ``BUILD_EXPORT_SET`` ( maps to  `<BUILD_INTERFACE>` ) and ``INSTALL_EXPORT_SET`` ( maps to `<INSTALL_INTERFACE>` ) keywords on commands such as :cmake:command:`rapids_find_package` and :cmake:command:`rapids_cpm_find`.
+Lets go over an example on how these come together to make dependency tracking for projects super easy:
+
+.. code-block:: cmake
+
+  rapids_find_package(Threads REQUIRED
+    BUILD_EXPORT_SET example-targets
+    INSTALL_EXPORT_SET example-targets
+    )
+  rapids_cpm_find(nlohmann_json 3.9.1
+    BUILD_EXPORT_SET example-targets
+    )
+
+  add_library(example src.cpp)
+  target_link_libraries(example 
+    PUBLIC Threads::Threads
+      $<BUILD_INTERFACE:nlohmann_json::nlohmann_json>
+    )
+  install(TARGETS example
+    DESTINATION lib
+    EXPORT example-targets
+    )                 
+
+  set(doc_string [=[Provide targets for the example library.]=])
+  rapids_export(INSTALL example
+    EXPORT_SET example-targets
+    NAMESPACE example::
+    DOCUMENTATION doc_string
+    )
+  rapids_export(BUILD example
+    EXPORT_SET example-targets
+    NAMESPACE example::
+    DOCUMENTATION doc_string
+    )
+
+
+Tracking find_package
+*********************
+
+The :cmake:command:`rapids_find_package(<PackageName>)` combines the :cmake:command:`find_package` command and dependency tracking.
+In this example we are recording that the `Threads` package is required by both the export set `example-targets` for both the install and build configuration.
+
+What this means is that when :cmake:command:`rapids_export(` is called a the `example-config.cmake` file is generated it will include the call 
+`find_dependency(Threads)`, removing the need for developers to mantain that dual source of truth.
+
+The :cmake:command:`rapids_find_package(<PackageName>)` command also supports non-required finds correctly. In those cases `rapids-cmake` only records
+the dependency when the underlying :cmake:command:`find_package` command is successful.
+
+It is common for projects to depend on dependencies that CMake doesn't have `Find<Package>` for. In those cases projects will have a custom
+`Find<Package>` that they need to use, and install for consumers. Rapids-cmake tries to help projects simplify this process with the commands
+:cmake:command:`rapids_find_generate_module` and :cmake:command:`rapids_export_package`.
+
+The :cmake:command:`rapids_find_generate_module` allows for projects to automatically generate a `Find<Package>` and encode via the `BUILD_EXPORT_SET`
+and `INSTALL_EXPORT_SET` commands when the generated module should also be installed and part of `CMAKE_MODULE_PATH` so that consumers can use it.
```

Comment:
```suggestion
The :cmake:command:`rapids_find_generate_module` allows projects to automatically generate a `Find<Package>` and encode via the `BUILD_EXPORT_SET`
and `INSTALL_EXPORT_SET` parameters when the generated module should also be installed and added to  `CMAKE_MODULE_PATH` so that consumers can use it.
```